beta-src/src/components/ui/WDMessageList.tsx

```diff
@@ -24,27 +23,53 @@ const WDMessageList: React.FC<WDMessageListProps> = function ({
   countryIDSelected,
   messagesEndRef,
 }): React.ReactElement {
-  const [viewport] = useViewport();
-  const device = getDevice(viewport);
-  const height = "350px";
+  const viewedPhaseState = useAppSelector(gameViewedPhase);
+
   const filteredMessages = messages.filter(
     (message) =>
       (message.fromCountryID === countryIDSelected ||
         message.toCountryID === countryIDSelected) &&
       (countryIDSelected === 0 || message.toCountryID !== 0), // public messages in public chat
   );
-  const messageComponents = filteredMessages.map((message: GameMessage) => (
-    <WDMessage
-      key={`${message.timeSent}:${message.fromCountryID}:${message.toCountryID}:${message.message}`}
-      message={message}
-      userCountry={userCountry}
-      allCountries={allCountries}
-    />
-  ));
+  const messagesByTurn: { [key: number]: React.ReactElement[] } = {};
+  filteredMessages.forEach((message: GameMessage) => {
+    if (!(message.turn in messagesByTurn)) {
+      messagesByTurn[message.turn] = [];
+    }
+    messagesByTurn[message.turn].push(
+      <WDMessage
+        key={`${message.timeSent}:${message.fromCountryID}:${message.toCountryID}:${message.message}`}
+        message={message}
+        userCountry={userCountry}
+        allCountries={allCountries}
+        viewedPhaseIdx={viewedPhaseState.viewedPhaseIdx}
+      />,
+    );
+  });
+  const messageTurnComponents = Object.entries(messagesByTurn).map(
+    ([turn, msgs]) => {
+      const psy = getPhaseSeasonYear(Number(turn), "Diplomacy");
```

Comment:
I'm surprised this isn't a lint error - `Number()` does some funny coercion things in the case of empty strings (`Number("")` returns `0`) and behaves strangely with strings that aren't decimal (it tries to guess the radix). Some implementations even treat strings with a leading `0` as octal.

It's better practice to use `Number.parseInt(turn, 10)` (with the radix explicit so that it won't mistreat leading zeroes). This will correctly return `NaN` if you pass an empty string.

It won't matter if you're careful to always pass decimal strings without leading zeroes, but if you accidentally pass the wrong thing, tracking down the bug can be hard. Better to always use `Number.parseInt(whateverVariable, 10)`

Reply:
Done. Yeah, the original sin here is that we haven't been disciplined about always treating numbers as numbers - some numbers are retrieved as strings from the server and stored that way, and it will take a careful refactor to fix this everywhere.